src/pip/_internal/build_env/_venv.py

```diff
@@ -0,0 +1,106 @@
+"""`venv`-based environment for package builds
+"""
+
+import os
+import sys
+import venv
+from types import TracebackType
+from typing import TYPE_CHECKING, Dict, Iterable, Optional, Set, Tuple, Type
+
+from pip._vendor.packaging.requirements import Requirement
+from pip._vendor.packaging.version import Version
+from pip._vendor.requests.certs import where
+
+from pip._internal.cli.spinners import open_spinner
+from pip._internal.metadata import get_environment
+from pip._internal.utils.subprocess import call_subprocess
+from pip._internal.utils.temp_dir import TempDirectory, tempdir_kinds
+
+from ._base import get_runnable_pip, iter_install_flags
+
+if TYPE_CHECKING:
+    from pip._internal.index.package_finder import PackageFinder
+
+
+class VenvBuildEnvironment:
+    """A build environment that does nothing."""
+
+    def __init__(self) -> None:
+        self._temp_dir = TempDirectory(
+            kind=tempdir_kinds.BUILD_ENV, globally_managed=True
+        )
+        self._venv = venv.EnvBuilder()
+        context = self._venv.ensure_directories(self._temp_dir.path)
+        self._venv.create(self._temp_dir.path)
+
+        # Copy-pasted from venv/__init__.py
+        if sys.platform == "win32":
+            libpath = os.path.join(
+                self._temp_dir.path,
+                "Lib",
+                "site-packages",
+            )
+        else:
+            libpath = os.path.join(
+                self._temp_dir.path,
+                "lib",
+                f"python{sys.version_info.major}.{sys.version_info.minor}",
+                "site-packages",
+            )
```

Comment:
Maybe we can use sysconfig here instead? New versions have a `venv` key for this specific use case, and we can fall back to the prefix scheme otherwise (or maybe limit this to 3.11+).

Reply:
TBH, I'd like to switch over to this in a regular/slightly-long deprecation cycle to let us kill the old isolation logic and apply pressure on redistributors/users to unbreak their Python distributions. But, I digress.

Regarding sysconfig/venv scheme: yea, I think we could but this is matching the current implementation realities across all versions, and isn't particularly complex or difficult to maintain. It's only feasible for this to change in a new Python version which we will have the ability to adapt to. :)